File: PowerOf4/solution.py
class Solution(object):
    def isPowerOfFour(self, num):
        """
        :type num: int
        :rtype: bool
        """
        return num != 0 and num & (num-1) == 0 and num & 1431655765 == num
        
    def isPowerOfFour2(self, num):
      return num & (num-1) == 0 and (num -1) % 3 == 0

File: PowerOf4/solution.java
class Solution {
  public boolean isPowerOf4(int n) {
    return this.isPowerOf2(n) && (n-1) % 4 == 3;
  }
  
  public boolean isPowerOf2(int n) {
    return n > 0 && n & (n-1) == 0;
  }
}
